src/basho_bench.erl

```diff
@@ -104,17 +109,12 @@ main(Args) ->
     ok = file:set_cwd(TestDir),
     log_dimensions(),
 
-    %% Run pre_hook for user code preconditions
-    run_pre_hook(),
-    %% Spin up the application
-    ok = basho_bench_app:start(),
-
-    %% Pull the runtime duration from the config and sleep until that's passed OR
-    %% the supervisor process exits
-    Mref = erlang:monitor(process, whereis(basho_bench_sup)),
-    DurationMins = basho_bench_config:get(duration),
-    wait_for_stop(Mref, DurationMins).
-
+    run_benchmark(),
+    MRef = erlang:monitor(process, whereis(basho_bench_run_sup)),
+    receive
+        {'DOWN', MRef, process, _Object, _Info} ->
+            ok
+    end.
```

Comment:
Seems like we should have this chunk of code available in another function, and have it be optional here.